compiler/rustc_codegen_ssa/src/sir.rs

```diff
@@ -103,9 +103,9 @@ impl SirFuncCx<'tcx> {
 
                 let inst_kind = instance.ty(tcx, ty::ParamEnv::reveal_all()).kind();
                 if !matches!(inst_kind, ty::FnDef(..)) {
-                    // We can't accept closures, as these have a hidden argument prepended for the
-                    // upvars and this breaks the invariant that Local(1) is the interpreter
-                    // context.
+                    // We can't accept anything that isn't a regular function. This includes
```

Comment:
I think we need to deemphasise the second sentence a bit. Something like "We reject anything that isn't a regular function to ensure that we don't encounter odd problems elsewhere. For example, closures prepend an argument for upvars which breaks the invariant that Local(1) is the interpreter context."